pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -57,6 +57,65 @@ func (r *ReconcileKubeDirectorCluster) syncCluster(
 		cr.Status = &kdv1.KubeDirectorClusterStatus{}
 		cr.Status.Roles = make([]kdv1.RoleStatus, 0)
 	}
+
+	// Check, if some statefulset is in the middle of upgrade/rollback process.
```

Comment:
All this new logic added to syncCluster should be moved AFTER the big defer func definition below, just in case for some reason we want/need to do an early return out of syncCluster.

Also this function is already (too) big, so new logic should go into a subroutine e.g. something like handleUpgrade.